### PR TITLE
feat(HACBS-2381): update service to use operator-toolkit

### PR DESCRIPTION
**Pull request by davidmogar**
The operator-toolkit has been already released and operator-goodies is deprecated.

### CODE REVIEW

**Review by kasemAlem — APPROVED** [review]

File: api/v1alpha1/webhooks/author/suite_test.go
Comment (johnbieren) [inline api/v1alpha1/webhooks/author/suite_test.go:1]: I don't see how a single change in this file is related to "update service to use operator-toolkit"

File: api/v1alpha1/webhooks/author/webhook.go
Comment (johnbieren) [inline api/v1alpha1/webhooks/author/webhook.go:1]: I don't see how a single change in this file is related to "update service to use operator-toolkit"

File: api/v1alpha1/webhooks/author/suite_test.go
Comment (davidmogar) [reply api/v1alpha1/webhooks/author/suite_test.go:1]: Using operator-toolkit comes with declaring webhooks in a different way. That is, moving webhooks into new directories. Those new directories need a suite for tests to run.

File: api/v1alpha1/webhooks/author/webhook.go
Comment (davidmogar) [reply api/v1alpha1/webhooks/author/webhook.go:1]: Using operator-toolkit comes with declaring webhooks in a different way. That is, moving webhooks into new directories. Those new directories need the webhook code using the toolkit functions.

File: conditions/conditions_suite_test.go
Comment (johnbieren) [inline conditions/conditions_suite_test.go:1]: This should be added to operator-toolkit before being removed

File: conditions/conditions_test.go
Comment (johnbieren) [inline conditions/conditions_test.go:1]: This should be added to operator-toolkit before being removed

File: api/v1alpha1/webhooks/author/webhook.go [hunk 1]
```diff
@@ -29,35 +30,34 @@ import (
...
-		},
-	})
+	return nil
 }
 
+// +kubebuilder:webhook:path=/mutate-appstudio-redhat-com-v1alpha1-author,mutating=true,failurePolicy=fail,sideEffects=None,groups=appstudio.redhat.com,resources=releases;releaseplans,verbs=create;update,versions=v1alpha1,name=mauthor.kb.io,admissionReviewVersions=v1
+
 // Handle creates an admission response for Release and ReleasePlan requests.
-func (a *authorWebhook) Handle(ctx context.Context, req admission.Request) admission.Response {
+func (w *Webhook) Handle(ctx context.Context, req admission.Request) admission.Response {
```
Comment (johnbieren) [inline api/v1alpha1/webhooks/author/webhook.go:55]: How is this change necessary for adding toolkit and not a rewrite?

File: api/v1alpha1/webhooks/author/webhook.go [hunk 2]
```diff
@@ -125,20 +125,20 @@ func (a *authorWebhook) handleReleasePlan(req admission.Request) admission.Respo
 		if releasePlan.GetLabels()[metadata.AttributionLabel] == "true" {
 			author := releasePlan.GetLabels()[metadata.AuthorLabel]
 
-			if oldReleasePlan.GetLabels()[metadata.AttributionLabel] != "true" || author == a.sanitizeLabelValue(req.UserInfo.Username) {
-				a.setAuthorLabel(req.UserInfo.Username, releasePlan)
+			if oldReleasePlan.GetLabels()[metadata.AttributionLabel] != "true" || author == w.sanitizeLabelValue(req.UserInfo.Username) {
+				w.setAuthorLabel(req.UserInfo.Username, releasePlan)
```
Comment (johnbieren) [inline api/v1alpha1/webhooks/author/webhook.go:129]: How is this change necessary for adding toolkit and not a rewrite?

File: api/v1alpha1/webhooks/author/webhook.go [hunk 3]
```diff
@@ -125,20 +125,20 @@ func (a *authorWebhook) handleReleasePlan(req admission.Request) admission.Respo
...
 			} else {
 				// Preserve previous author if the new author does not match the user making the change
-				a.setAuthorLabel(oldReleasePlan.GetLabels()[metadata.AuthorLabel], releasePlan)
+				w.setAuthorLabel(oldReleasePlan.GetLabels()[metadata.AuthorLabel], releasePlan)
 			}
 		}
 	}
 
-	return a.patchResponse(req.Object.Raw, releasePlan)
+	return w.patchResponse(req.Object.Raw, releasePlan)
```
Comment (johnbieren) [inline api/v1alpha1/webhooks/author/webhook.go:137]: How is this change necessary for adding toolkit and not a rewrite?

File: api/v1alpha1/webhooks/author/webhook.go [hunk 4]
```diff
@@ -125,20 +125,20 @@ func (a *authorWebhook) handleReleasePlan(req admission.Request) admission.Respo
...
 }
 
 // patchResponse returns an admission response that patches the passed raw object to be the passed object.
-func (a *authorWebhook) patchResponse(raw []byte, object client.Object) admission.Response {
+func (w *Webhook) patchResponse(raw []byte, object client.Object) admission.Response {
```
Comment (johnbieren) [inline api/v1alpha1/webhooks/author/webhook.go:141]: How is this change necessary for adding toolkit and not a rewrite?

File: api/v1alpha1/webhooks/author/webhook.go [hunk 5]
```diff
@@ -148,18 +148,18 @@ func (a *authorWebhook) patchResponse(raw []byte, object client.Object) admissio
 }
 
 // setAuthorLabel returns the passed object with the author label added.
-func (a *authorWebhook) setAuthorLabel(username string, obj client.Object) {
+func (w *Webhook) setAuthorLabel(username string, obj client.Object) {
```
Comment (johnbieren) [inline api/v1alpha1/webhooks/author/webhook.go:151]: How is this change necessary for adding toolkit and not a rewrite?

File: api/v1alpha1/webhooks/author/webhook.go [hunk 6]
```diff
@@ -148,18 +148,18 @@ func (a *authorWebhook) patchResponse(raw []byte, object client.Object) admissio
...
 	labels := make(map[string]string)
 	if obj.GetLabels() != nil {
 		labels = obj.GetLabels()
 	}
 
-	labels[metadata.AuthorLabel] = a.sanitizeLabelValue(username)
+	labels[metadata.AuthorLabel] = w.sanitizeLabelValue(username)
```
Comment (johnbieren) [inline api/v1alpha1/webhooks/author/webhook.go:157]: How is this change necessary for adding toolkit and not a rewrite?

File: api/v1alpha1/webhooks/author/webhook.go [hunk 7]
```diff
@@ -148,18 +148,18 @@ func (a *authorWebhook) patchResponse(raw []byte, object client.Object) admissio
...
 	obj.SetLabels(labels)
 }
 
 // sanitizeLabelValue takes a username and returns it in a form appropriate to use as a label value.
-func (a *authorWebhook) sanitizeLabelValue(username string) string {
+func (w *Webhook) sanitizeLabelValue(username string) string {
```
Comment (johnbieren) [inline api/v1alpha1/webhooks/author/webhook.go:162]: How is this change necessary for adding toolkit and not a rewrite?

File: syncer/syncer.go [hunk 8]
```diff
@@ -28,16 +28,16 @@ import (
 type Syncer struct {
 	client client.Client
 	ctx    context.Context
-	logger logr.Logger
+	logger *logr.Logger
```
Comment (johnbieren) [inline syncer/syncer.go:31]: How is this change necessary for adding toolkit and not a rewrite?

File: syncer/syncer.go [hunk 9]
```diff
@@ -28,16 +28,16 @@ import (
...
 }
 
 // NewSyncer creates a new Syncer with the given client and logger.
-func NewSyncer(client client.Client, logger logr.Logger) *Syncer {
+func NewSyncer(client client.Client, logger *logr.Logger) *Syncer {
```
Comment (johnbieren) [inline syncer/syncer.go:35]: How is this change necessary for adding toolkit and not a rewrite?

File: syncer/syncer.go [hunk 10]
```diff
@@ -28,16 +28,16 @@ import (
...
 	return NewSyncerWithContext(client, logger, context.TODO())
 }
 
 // NewSyncerWithContext creates a new Syncer with the given client, logger and context.
-func NewSyncerWithContext(client client.Client, logger logr.Logger, ctx context.Context) *Syncer {
+func NewSyncerWithContext(client client.Client, logger *logr.Logger, ctx context.Context) *Syncer {
```
Comment (johnbieren) [inline syncer/syncer.go:40]: How is this change necessary for adding toolkit and not a rewrite?

File: api/v1alpha1/webhooks/author/webhook.go [hunk 11]
```diff
@@ -29,35 +30,34 @@ import (
 	admissionv1 "k8s.io/api/admission/v1"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	ctrlWebhook "sigs.k8s.io/controller-runtime/pkg/webhook"
 	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
 )
 
-// authorWebhook describes the data structure for the author webhook
-type authorWebhook struct {
```
[the remark this answers is not in the document]
Comment (davidmogar) [reply api/v1alpha1/webhooks/author/webhook.go:37]: Consistency across webhooks.

File: api/v1alpha1/webhooks/author/webhook.go [hunk 12]
```diff
@@ -29,35 +30,34 @@ import (
...
+type Webhook struct {
 	client client.Client
-	log    *logr.Logger
+	log    logr.Logger
 }
 
-// +kubebuilder:webhook:path=/mutate-appstudio-redhat-com-v1alpha1-author,mutating=true,failurePolicy=fail,sideEffects=None,groups=appstudio.redhat.com,resources=releases;releaseplans,verbs=create;update,versions=v1alpha1,name=mauthor.kb.io,admissionReviewVersions=v1
+// Register registers the webhook with the passed manager and log.
+func (w *Webhook) Register(mgr ctrl.Manager, log *logr.Logger) error {
+	w.client = mgr.GetClient()
```
[the remark this answers is not in the document]
Comment (davidmogar) [reply api/v1alpha1/webhooks/author/webhook.go:44]: Consistency. The information to pass has to be carried into the webhook struct and given that it's received as a receiver, the new code is better.

File: api/v1alpha1/webhooks/author/webhook.go [hunk 13]
```diff
@@ -29,35 +30,34 @@ import (
...
 
+// +kubebuilder:webhook:path=/mutate-appstudio-redhat-com-v1alpha1-author,mutating=true,failurePolicy=fail,sideEffects=None,groups=appstudio.redhat.com,resources=releases;releaseplans,verbs=create;update,versions=v1alpha1,name=mauthor.kb.io,admissionReviewVersions=v1
+
 // Handle creates an admission response for Release and ReleasePlan requests.
-func (a *authorWebhook) Handle(ctx context.Context, req admission.Request) admission.Response {
+func (w *Webhook) Handle(ctx context.Context, req admission.Request) admission.Response {
 	switch req.Kind.Kind {
 	case "Release":
-		return a.handleRelease(req)
+		return w.handleRelease(req)
```
[the remark this answers is not in the document]
Comment (davidmogar) [reply api/v1alpha1/webhooks/author/webhook.go:58]: Consistency.

File: api/v1alpha1/webhooks/author/webhook.go [hunk 14]
```diff
@@ -29,35 +30,34 @@ import (
...
 	case "ReleasePlan":
-		return a.handleReleasePlan(req)
+		return w.handleReleasePlan(req)
```
[the remark this answers is not in the document]
Comment (davidmogar) [reply api/v1alpha1/webhooks/author/webhook.go:60]: Consistency.

File: api/v1alpha1/webhooks/author/webhook.go [hunk 15]
```diff
@@ -67,8 +67,8 @@ func (a *authorWebhook) Handle(ctx context.Context, req admission.Request) admis
 // handleRelease takes an incoming admission request and returns an admission response. Create requests
 // add an author label with the current user. Update requests are rejected if the author label is being
 // modified. All other requests are accepted without action.
-func (a *authorWebhook) handleRelease(req admission.Request) admission.Response {
```
[the remark this answers is not in the document]
Comment (davidmogar) [reply api/v1alpha1/webhooks/author/webhook.go:70]: Consistency.

File: api/v1alpha1/webhooks/author/webhook.go (hunk 2 above)
Comment (davidmogar) [reply api/v1alpha1/webhooks/author/webhook.go:129]: Consistency.

File: api/v1alpha1/webhooks/author/webhook.go [hunk 16]
```diff
@@ -125,20 +125,20 @@ func (a *authorWebhook) handleReleasePlan(req admission.Request) admission.Respo
 		if releasePlan.GetLabels()[metadata.AttributionLabel] == "true" {
 			author := releasePlan.GetLabels()[metadata.AuthorLabel]
 
-			if oldReleasePlan.GetLabels()[metadata.AttributionLabel] != "true" || author == a.sanitizeLabelValue(req.UserInfo.Username) {
-				a.setAuthorLabel(req.UserInfo.Username, releasePlan)
+			if oldReleasePlan.GetLabels()[metadata.AttributionLabel] != "true" || author == w.sanitizeLabelValue(req.UserInfo.Username) {
```
[the remark this answers is not in the document]
Comment (davidmogar) [reply api/v1alpha1/webhooks/author/webhook.go:128]: Consistency.

File: api/v1alpha1/webhooks/author/webhook.go [hunk 17]
```diff
@@ -113,10 +113,10 @@ func (a *authorWebhook) handleReleasePlan(req admission.Request) admission.Respo
 	switch req.AdmissionRequest.Operation {
 	case admissionv1.Create:
 		if releasePlan.GetLabels()[metadata.AttributionLabel] == "true" {
-			a.setAuthorLabel(req.UserInfo.Username, releasePlan)
+			w.setAuthorLabel(req.UserInfo.Username, releasePlan)
```
[the remark this answers is not in the document]
Comment (davidmogar) [reply api/v1alpha1/webhooks/author/webhook.go:116]: Consistency.

File: api/v1alpha1/webhooks/author/webhook.go [hunk 18]
```diff
@@ -99,8 +99,8 @@ func (a *authorWebhook) handleRelease(req admission.Request) admission.Response
 // attribution label is set to true, the current user is set as the author. If the attribution label
 // is false, the author label is removed. The only exception is if the attribution label remains true
 // during an update and the author value is not modified, the previous author label remains.
-func (a *authorWebhook) handleReleasePlan(req admission.Request) admission.Response {
-	releasePlan := &ReleasePlan{}
+func (w *Webhook) handleReleasePlan(req admission.Request) admission.Response {
```
[the remark this answers is not in the document]
Comment (davidmogar) [reply api/v1alpha1/webhooks/author/webhook.go:102]: Consistency.

File: api/v1alpha1/webhooks/author/webhook.go [hunk 19]
```diff
@@ -77,12 +77,12 @@ func (a *authorWebhook) handleRelease(req admission.Request) admission.Response
 	switch req.AdmissionRequest.Operation {
 	case admissionv1.Create:
 		if release.GetLabels()[metadata.AutomatedLabel] != "true" {
-			a.setAuthorLabel(req.UserInfo.Username, release)
+			w.setAuthorLabel(req.UserInfo.Username, release)
 		}
 
-		return a.patchResponse(req.Object.Raw, release)
+		return w.patchResponse(req.Object.Raw, release)
```
[the remark this answers is not in the document]
Comment (davidmogar) [reply api/v1alpha1/webhooks/author/webhook.go:83]: Consistency.

File: api/v1alpha1/webhooks/author/webhook.go [hunk 20]
```diff
@@ -77,12 +77,12 @@ func (a *authorWebhook) handleRelease(req admission.Request) admission.Response
 	switch req.AdmissionRequest.Operation {
 	case admissionv1.Create:
 		if release.GetLabels()[metadata.AutomatedLabel] != "true" {
-			a.setAuthorLabel(req.UserInfo.Username, release)
+			w.setAuthorLabel(req.UserInfo.Username, release)
```
[the remark this answers is not in the document]
Comment (davidmogar) [reply api/v1alpha1/webhooks/author/webhook.go:80]: Consistency.

File: api/v1alpha1/webhooks/author/webhook.go [hunk 21]
```diff
@@ -125,20 +125,20 @@ func (a *authorWebhook) handleReleasePlan(req admission.Request) admission.Respo
...
 			author := releasePlan.GetLabels()[metadata.AuthorLabel]
 
-			if oldReleasePlan.GetLabels()[metadata.AttributionLabel] != "true" || author == a.sanitizeLabelValue(req.UserInfo.Username) {
-				a.setAuthorLabel(req.UserInfo.Username, releasePlan)
+			if oldReleasePlan.GetLabels()[metadata.AttributionLabel] != "true" || author == w.sanitizeLabelValue(req.UserInfo.Username) {
+				w.setAuthorLabel(req.UserInfo.Username, releasePlan)
 			} else {
 				// Preserve previous author if the new author does not match the user making the change
-				a.setAuthorLabel(oldReleasePlan.GetLabels()[metadata.AuthorLabel], releasePlan)
+				w.setAuthorLabel(oldReleasePlan.GetLabels()[metadata.AuthorLabel], releasePlan)
```
[the remark this answers is not in the document]
Comment (davidmogar) [reply api/v1alpha1/webhooks/author/webhook.go:132]: Consistency.

File: api/v1alpha1/webhooks/author/webhook.go (hunk 7 above)
Comment (davidmogar) [reply api/v1alpha1/webhooks/author/webhook.go:162]: Consistency.

File: syncer/syncer.go (hunk 8 above)
Comment (davidmogar) [reply syncer/syncer.go:31]: Because this comes from the adapter and we are following how the new adapter should be defined.

File: api/v1alpha1/webhooks/author/webhook.go (hunk 11 above)
[the remark this answers is not in the document]
Comment (johnbieren) [reply api/v1alpha1/webhooks/author/webhook.go:37]: Then make a new PR called "consistency across webhooks" and add the hundreds of lines of changes related to it. Don't sneak it in this PR
